File: week-2/longest-word-in-dictionary/longest-word-in-dictionary.js
/**
 * @param {string[]} words
 * @return {string}
 */
var longestWord = function(words) {
    const hashtable = { '': true };
    let longest = '';
    words.sort();
    for (let word of words) {
        if (word.slice(0, -1) in hashtable){
            hashtable[word] = true;
            if (word.length > longest.length)
                longest = word;
        }
    }
    
    return longest;
};

File: week-2/longest-word-in-dictionary/longest-word-in-dictionary.py
class Solution:
    def longestWord(self, words: List[str]) -> str:
        hashtable = { '': True }
        longest = ''
        words.sort()
        for word in words:
            if word[:-1] in hashtable:
                hashtable[word] = True
                if len(word) > len(longest):
                    longest = word
                    
        return longest
